gamestonk_terminal/stocks/options/yfinance_model.py

```diff
@@ -117,9 +117,54 @@ def generate_data(
     return x_vals, before, []
 
 
-def get_price(ticker) -> float:
-    """Get current price for a given ticker"""
+def get_price(ticker: str) -> float:
+    """Get current price for a given ticker
+
+    Parameters
+    ----------
+    ticker : str
+        The ticker to get the price for
+
+    Returns
+    ----------
+    price : float
+        The price of the ticker
+    """
     ticker_yahoo = yf.Ticker(ticker)
     data = ticker_yahoo.history()
     last_quote = data.tail(1)["Close"].iloc[0]
     return last_quote
+
```

Comment:
An actual question:  is this faster than doing yf.Ticker().info["regularMarketPrice"]?

Reply:
Shocking results here. I ran code because I was generally curious and my way was 20 - 30 times faster. Not sure why.

`import yfinance as yf
from datetime import datetime

print("Running time for 10 requests")
print("Colin's Method:")

colin_start = datetime.now()
for _ in range(10):
    ticker_yahoo = yf.Ticker("TSLA")
    data = ticker_yahoo.history()
    last_quote = data.tail(1)["Close"].iloc[0]
colin_end = datetime.now()

colin_diff = (colin_end - colin_start)
print(colin_diff.total_seconds() * 1000)

print("\nJame's Method:")

james_start = datetime.now()
for _ in range(10):
    last_quote = yf.Ticker("TSLA").info["regularMarketPrice"]
james_end = datetime.now()

james_diff = (james_end - james_start)
print(james_diff.total_seconds() * 1000)`

Reply:
So I actually did this previously.  Same results:

https://user-images.githubusercontent.com/18151143/143073050-794beea1-524b-423f-a914-f1ef32207cbc.png

My guess is getting all the dictionary entries takes longer than getting the time series.

Reply:
@colin99d FYI, you can use timeit (https://docs.python.org/3/library/timeit.html) for this type of assessment

This doesn't surprise me though @jmaslek, as this is the reason why I mention having the last price from when we do `load` rather than with `quote`.

Reply:
Thats my guess too.